bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -394,36 +413,39 @@ private void updatePropsFromJsonObject(MiIoSendCommand response) {
 
     private void updateChannel(@Nullable MiIoBasicChannel basicChannel, String param, JsonElement value) {
         JsonElement val = value;
-        if (basicChannel != null) {
-            if (basicChannel.getTransfortmation() != null) {
-                JsonElement transformed = Conversions.execute(basicChannel.getTransfortmation(), val);
-                logger.debug("Transformed with '{}': {} {} -> {} ", basicChannel.getTransfortmation(),
-                        basicChannel.getFriendlyName(), val, transformed);
-                val = transformed;
+        if (basicChannel == null) {
+            logger.debug("Channel not found for {}", param);
+            return;
+        }
+        final String transformation = basicChannel.getTransfortmation();
+        if (transformation != null) {
+            JsonElement transformed = Conversions.execute(transformation, val);
+            logger.debug("Transformed with '{}': {} {} -> {} ", transformation, basicChannel.getFriendlyName(), val,
+                    transformed);
+            val = transformed;
+        }
+        try {
+            if (basicChannel.getType().equals("Number")) {
```

Comment:
looks like a good place to use a switch statement.

Reply:
yes, that makes the code cleaner... done